src/app/angular/cdk-tree/cdk-tree.component.ts

```diff
@@ -75,21 +11,24 @@ interface ExampleFlatNode {
 })
 export class CdkTreeComponent {
   static selector = 'appCdkTree';
+  private treeData: ExampleFlatNode[] = this.cdkTreeService.getTreeData();
   treeControl = new FlatTreeControl<ExampleFlatNode>(
     node => node.level,
     node => node.expandable
   );
 
-  dataSource = new ArrayDataSource(TREE_DATA);
+  constructor(private cdkTreeService: CdkTreeService) {}
+
+  dataSource = new ArrayDataSource(this.treeData);
 
   hasChild = (_: number, node: ExampleFlatNode) => node.expandable;
 
   getParentNode(node: ExampleFlatNode) {
-    const nodeIndex = TREE_DATA.indexOf(node);
+    const nodeIndex = this.treeData.indexOf(node);
 
     for (let i = nodeIndex - 1; i >= 0; i--) {
-      if (TREE_DATA[i].level === node.level - 1) {
-        return TREE_DATA[i];
+      if (this.treeData[i].level === node.level - 1) {
+        return this.treeData[i];
       }
     }
```

Comment:
This can be simplified with the use of `for...of`:

```ts
    for (const otherNode of this.treeData) {
      if (otherNode.level === node.level - 1) {
        return otherNode;
      }
    }
```

Reply:
This, unfortunately, will not work the same way. The original loop iterates from the given node upwards and checks if there is a parent, if so, it renders it. Your loop looks for a parent and renders it, but with the leaf omitted. 
Unfortunately, CDK Tree is not working properly, 
<img width="240" alt="image" src="https://user-images.githubusercontent.com/5073455/223482592-75243bcb-6e94-49a2-871e-dcbcbe3da2ad.png">

Reply:
Sorry, I don't know what I was thinking :grin:
I obviously missed that we are not iterating through the array from the beginning 🤦‍♂️